edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java

```diff
@@ -1,7 +1,149 @@
 package fr.univtln.lhd.facade;
 
+import fr.univtln.lhd.model.entities.dao.slots.SlotDAO;
+import fr.univtln.lhd.model.entities.slots.Group;
+import fr.univtln.lhd.model.entities.users.Student;
+import fr.univtln.lhd.model.entities.slots.Slot;
+import lombok.extern.slf4j.Slf4j;
+import org.threeten.extra.Interval;
+
+import java.sql.SQLException;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
- * Core classe of our software
+ * facade of the entities/dao and the ihm
  */
+@Slf4j
 public class Schedule {
+    private static final ZoneId TIME_ZONE = ZoneOffset.systemDefault();
+    private static final Schedule schedule = new Schedule();
+
+    /**
+     * Take a student and two LocalDate(start/end) and return every slot that the student is
+     * in in the timerange of the two LocalDate
+     * The student must be persisted or will return an empty array
+     *
+     * @param student   a persisted Student
+     * @param timeStart the start of the timerange
+     * @param timeEnd   the end of the timerange
+     * @return a List of slot that is in the timerange and with the Groups of the Student
+     */
+    public static List<Slot> getSchedule(Student student, LocalDate timeStart, LocalDate timeEnd) {
+        Interval timerange = schedule.getIntervalOf(timeStart, timeEnd);
+        return Schedule.getSchedule(student, timerange);
+    }
+
+
+    /**
+     * Take a student and an Interval timerange and return every slot within the timerange
+     * where the student is in
+     *
+     * @param student   a persisted Student
+     * @param timerange An Interval
+     * @return a List of slot that is in the timerange and with the Groups of the Student
+     */
+    public static List<Slot> getSchedule(Student student, Interval timerange) {
+        List<Slot> slotList = new ArrayList<>();
+        if (student.getId() < 0) {
+            return slotList;
+        }//Might be an edge case
+        SlotDAO dao = SlotDAO.getInstance();
+        for (Group group :
+                student.getStudendGroup()) {
```

Comment:
careful, might conflict with latest changes on student entity

Reply:
Should not I only use the ID of group